src/NHibernate/Driver/DriverExtensions.cs

```diff
@@ -1,16 +1,63 @@
+using System.Data;
 using System.Data.Common;
 using NHibernate.AdoNet;
 using NHibernate.SqlTypes;
-using NHibernate.Type;
 
 namespace NHibernate.Driver
 {
-	internal static class DriverExtensions
+	public static class DriverExtensions
 	{
+		private static readonly INHibernateLogger Log = NHibernateLogger.For(typeof(DriverExtensions));
+		// No threading protection on this member: we do not care, at worst it will cause some duplicated warning.
+		private static bool _hasWarnedForMissingBeginTransaction;
+
 		internal static void AdjustParameterForValue(this IDriver driver, DbParameter parameter, SqlType sqlType, object value)
 		{
 			var adjustingDriver = driver as IParameterAdjuster;
 			adjustingDriver?.AdjustParameterForValue(parameter, sqlType, value);
 		}
+
+		// 6.0 TODO: merge into IDriver
+		/// <summary>
+		/// Begin an ADO <see cref="DbTransaction" />.
+		/// </summary>
+		/// <param name="driver">The driver.</param>
+		/// <param name="isolationLevel">The isolation level requested for the transaction.</param>
+		/// <param name="connection">The connection on which to start the transaction.</param>
+		/// <returns>The started <see cref="DbTransaction" />.</returns>
+		public static DbTransaction BeginTransaction(this IDriver driver, IsolationLevel isolationLevel, DbConnection connection)
+		{
+			if (driver is DriverBase driverBase)
+			{
+				return driverBase.BeginTransaction(isolationLevel, connection);
+			}
+
+			// Use reflection for supporting custom drivers.
+			var beginMethod = driver.GetType().GetMethod(
```

Comment:
The method is usually heavily used, so this needs to be cached at least into a compiled delegate (`MethodBase.Invoke` is bloody [slow](http://www.thejoyofcode.com/Performance_of_Method.Invoke_vs_a_Delegate.aspx)). However, I would encourage just remove the reflection from here.

Reply:
I do understand that this is "not common" to have a `IDriver` which does not extend `DriverBase`, but if someone does have this situation - they will suffer a lot.

Reply:
A cached compiled delegate would have to be done per driver type, something with a concurrent dictionary, and a GetOrAdd which would call the method if the driver has it, or call the inline implementation. At least it would handle more properly the warning. But well, it seems not worth it to do all this just for being slightly nicer by giving a hint for custom implementers, if they check NHibernate log, about getting ready for 6.0.